pytest_mergify/ci_insights.py

```diff
@@ -270,3 +374,54 @@ def mark_test_as_quarantined_if_needed(self, item: _pytest.nodes.Item) -> bool:
             return True
 
         return False
+
+
+def _select_affordable_tests(
+    budget_duration: datetime.timedelta,
+    test_durations: typing.Dict[str, datetime.timedelta],
+) -> typing.Dict[str, datetime.timedelta]:
+    """
+    Select tests that can be retried within the given budget.
+
+    This ensures we don't select tests that would exceed our time constraints
+    even with the minimum number of retries.
+    """
+
+    result = {}
+    for test, duration in test_durations.items():
+        if duration * _MIN_TEST_RETRY_COUNT <= budget_duration / len(test_durations):
+            result[test] = duration
```

Comment:
what about:
```suggestion
            yield test, duration
```

(maybe more for a future iteration)